### PR TITLE
Add base augmentation layer for 3D preception.

### PR DESCRIPTION
**Pull request by lengzq**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by lengzq** [comment]
@tanzhenyu

**Comment by bhack** [comment]
With all the performance gap we had with https://github.com/keras-team/keras-cv/issues/581#issuecomment-1271579684 and the other related KPL tickets are we sure that we want to go ahead with a within the batch augmentation policy?

**Comment by tanzhenyu** [comment]
> #581 (comment)

So quickly looking over the issue, do you mean `vectorized_map` is the main reason for performance gap?

**Comment by bhack** [comment]
It is not strictly or exclusively the vectorized map but I think our within the batch randomizattion policy when the underline TF ops  that we want to use doen't have a batched argument so also a working vectorized map can "miracles" (this was quite frequent with `tf.image` ops in KLP)

Also the main problem is that after many months and users tickets we haven't done any step ahead, also with the `vectorized_map` API owners involved in TF, in this analysis. On the other side it seems that we don't have any specific paper support point on the superior performance or faster convergence of the within the batch randomization policy.

As I know the only thing that we have for the batch augmentation policy with a support paper is:
https://github.com/keras-team/keras-cv/issues/372#issuecomment-1110827256

**Comment by tanzhenyu** [comment]
> It is not strictly or exclusively the vectorized map but I think our within the batch randomizattion policy when the underline TF ops that we want to use don't have batched argument (this was quite frequent with `tf.image` ops in KLPs) . So also a working vectorized map converter cannot do "miracles".
> 
> Also the main problem is that after many months and users tickets we haven't done any step ahead, also with the `vectorized_map` API owners involved in TF, in this analysis. On the other side it seems that we don't have any specific paper/support point related to the superior performance or faster convergence of the within the batch randomization policy.
> 
> As I know the only thing that we have for the batch augmentation policy with a support paper is: [#372 (comment)](https://github.com/keras-team/keras-cv/issues/372#issuecomment-1110827256)

then let's just keep using `tf.map_fn` and other batched implementations

**Comment by bhack** [comment]
> then let's just keep using tf.map_fn and other batched implementations

I think you missed the point. It will be probably slow as the vectorized_map fallback.

The problem is that if we want randomize over every single batch sample when the TF ops use in the preprocessing function don't accept vectorized arguments we will go to heavily impact  the training process performance.  
The point is: for what? You haven't expose any empirical experiment or paper that sustain this big performance gap compensating it with an higher convergence rate.

At least https://github.com/keras-team/keras-cv/issues/372#issuecomment-1110827256 could limit the performance gap a we could use at least a TF ops in "sub-batches".

File: keras_cv/layers/preprocessing3d/base_augmentation_layer_3d_test.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d.base_augmentation_layer_3d import (
+    BaseAugmentationLayer3D,
+)
+
+POINT_CLOUDS = "point_clouds"
```
Comment (tanzhenyu) [inline keras_cv/layers/preprocessing3d/base_augmentation_layer_3d_test.py:21]: just use BaseAugmentationLayer3D.POINT_CLOUDS?

**Review by tanzhenyu — APPROVED** [review]
LGTM, some minor comments.

**Comment by bhack** [comment]
If you prefer I could delete my comments so the PR is more readable. Let me know.

**Comment by tanzhenyu** [comment]
> As it is approved, If you prefer I could delete my comments so the PR is more readable. Let me know.

It's ok, I think both the contributor and I are aware of the current status

**Comment by bhack** [comment]
> It's ok, I think both the contributor and I are aware of the current status

A more explicit rollout is at https://github.com/keras-team/keras-cv/issues/581#issuecomment-1304554115 if you want to subscribe.

**Pull request merged**